File: LeetCode/Study Plan/Practice Problems/Easy/Reverse Vowels Of A String/Cpp/reverse_vowels_of_a_string.cc
#include <iostream>
#include <vector>

typedef std::vector<char> vc;

std::string reverse_vowels_of_a_string(std::string word) {
    vc vow;
    int k =0;
    for (int i=0; i<word.length(); i++)
        if (word[i] =='a' || word[i] == 'e' || word[i] == 'i' || word[i] == 'o' || word[i] =='u' || word[i]=='A' || word[i]== 'E' || word[i] == 'I' || word[i] == 'O' || word[i] == 'U') vow.push_back(word[i]);
    reverse(vow.begin(), vow.end());
    for (int i=0;i<word.length();i++) {
        if (word[i] =='a' || word[i] == 'e' || word[i] == 'i' || word[i] == 'o' || word[i] =='u' || word[i]=='A' || word[i]== 'E' || word[i] == 'I' || word[i] == 'O' || word[i] == 'U') {
            word[i] = vow[k];
            k++;
        }
    }
    return word;
}

int main () {
    std::cout<<reverse_vowels_of_a_string("hello")<<"\n";
}

File: LeetCode/Study Plan/Practice Problems/Easy/Reverse Vowels Of A String/Go/reverse_vowels_of_a_string.go
package main

import "fmt"

func reverse_vowels_of_a_string(s string) string {
	vow := []byte{}
	str := []byte(s)
	n := len(str)
	for i := 0; i < n; i++ {
		if str[i] == 'a' || str[i] == 'e' || str[i] == 'o' || str[i] == 'u' || str[i] == 'i' || str[i] == 'A' || str[i] == 'E' || str[i] == 'I' || str[i] == 'O' || str[i] == 'U' {
			vow = append(vow, str[i])
		}
	}
	reverse(&vow)
	k := 0
	for i := 0; i < n; i++ {
		if str[i] == 'a' || str[i] == 'e' || str[i] == 'o' || str[i] == 'u' || str[i] == 'i' || str[i] == 'A' || str[i] == 'E' || str[i] == 'I' || str[i] == 'O' || str[i] == 'U' {
			str[i] = vow[k]
			k++
		}
	}
	return string(str)
}

func reverse(vow *[]byte) {
	n := len(*vow)
	for i := 0; i < n/2; i++ {
		(*vow)[i], (*vow)[n-1-i] = (*vow)[n-1-i], (*vow)[i]
	}
}

func main() {
	fmt.Println(reverse_vowels_of_a_string("hello"))
}
